go-controller/pkg/clustermanager/pod/allocator_test.go

```diff
@@ -485,10 +494,45 @@ func TestPodAllocator_reconcileForNAD(t *testing.T) {
 			ipam:            true,
 			expectIPRelease: true,
 		},
+		{
+			name: "Pod with primary network NSE, expect event and error",
+			args: args{
+				new: &testPod{
+					scheduled: true,
+					network: &nadapi.NetworkSelectionElement{
+						Namespace: "namespace",
+						Name:      "nad",
+					},
+				},
+				nads: []*nadapi.NetworkAttachmentDefinition{
+					ovntest.GenerateNAD("surya", "nad", "namespace",
+						types.Layer3Topology, "100.128.0.0/16", types.NetworkRolePrimary),
+				},
+			},
+			role:         types.NetworkRolePrimary,
+			expectError:  "failed to get NAD to network mapping: unexpected primary network \"\" specified with a NetworkSelectionElement &{Name:nad Namespace:namespace IPRequest:[] MacRequest: InfinibandGUIDRequest: InterfaceRequest: PortMappingsRequest:[] BandwidthRequest:<nil> CNIArgs:<nil> GatewayRequest:[] IPAMClaimReference:}",
+			expectEvents: []string{"Warning ErrorAllocatingPod unexpected primary network \"\" specified with a NetworkSelectionElement &{Name:nad Namespace:namespace IPRequest:[] MacRequest: InfinibandGUIDRequest: InterfaceRequest: PortMappingsRequest:[] BandwidthRequest:<nil> CNIArgs:<nil> GatewayRequest:[] IPAMClaimReference:}"},
+		},
+		{
+			name: "Pod with primary network, expect event and error",
```

Comment:
this is a test for multiple primary nets in a namespace, right ? 

let's get the name right :) 

```suggestion
			name: "Pod on a namespace with multiple primary networks; expect event and error",
```